File: Power of That Base?/is_power_of_base.py
from math import log

def is_power_of_base(n, base):
    power = log(n, base)
    return int(power) == power

if __name__ == "__main__":
    n = int(input("Enter a number: "))
    base = int(input("Enter a base: "))
    print("{} is{}a power of {}".format(n, " " if is_power_of_base(n, base) else " NOT ", base))

File: Power of That Base?/IsPowerOfBase.java
import java.util.*;
import java.lang.Math.*;

class Main
{
    public static boolean isPowerOfBase(int n, int base) {
        double power = Math.log(n) / Math.log(base);
        return Math.floor(power) == Math.ceil(power);
    }


    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.println("Enter number");
        int number = scanner.nextInt();

        System.out.println("Enter base");
        int base = scanner.nextInt();

        System.out.println(isPowerOfBase(number, base));
    }
}